package/README.md

```diff
@@ -0,0 +1,59 @@
+# tlg.catalog.pkg
+
+This is a dummy package used for testing purposes. For the actual content please navigate to the [book directory](../book).
+
+The majority of the logic resides in the `tests/` subdirectory, while the package itself remains minimal.
+The main purpose of it is to reuse state of art GitHub Actions and other CI/CD tools for package development on the book content.
+This also includes `testthat2` snapshot testing for regression tests.
+
+## Regression tests
+
+### Overview
+
+During tests, as a part of [test setup](./tests/testthat/setup.R), the whole book will be re-rendered.
+This is possible because the content of the book is symlinked to the tests directory (symbolic link of `../book` as a `tests/testthat/_book`).
+There are dedicated `knitr` hooks and templates available and if used (see below section of how to use them), they will save relevant output objects as `.Rds` files.
+During actual testing, these files are loaded and compared with the expected ones using `testthat2` snapshot testing.
+
+A _standard_ render will be unaffected. These hooks will be used only during testing.
+
+### How to use `knitr` hooks and templates?
+
+As an article author, you need to (i) register the hooks, (ii) use them where appropriate and (iii) save the results.
+
+* At the beggining of the article, add the following: `{{< include ../../test-utils/envir_hook.qmd >}}`. It will register the `knitr` hooks for further use.
+* For code chunks without output - do nothing.
+* For code chunks with output:
```

Comment:
It would be nice to break lines before examples it that's possible.

Reply:
Thanks. I will check this one. Here, I need to comply with markdownlint which is tricky for lists

Reply:
I know :\ It's not crucial.

Reply:
ok done